xarray/core/dataarray.py

```diff
@@ -2132,6 +2132,50 @@ def rank(self, dim, pct=False, keep_attrs=False):
         ds = self._to_temp_dataset().rank(dim, pct=pct, keep_attrs=keep_attrs)
         return self._from_temp_dataset(ds)
 
+    def rolling_window(self, dim, window, window_dim, center=True):
```

Comment:
Can we make this a `Rolling` method instead? Maybe `construct` or `build` would be a good name for the method?

The API could look something like: `da.rolling(b=3).construct('window_dim')`

That helps keep the main DataArray/Dataset namespace a little more organized.

Reply:
How about `to_dataarray` and `to_dataset`?
Personally, `construct` sounds something necessary to compute rolling method.